From c8c774712a2aaca513e8bee4cc0d62860a3ba0f2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pekka=20J=C3=A4=C3=A4skel=C3=A4inen?=
 <pekka.jaaskelainen@intel.com>
Date: Thu, 15 Dec 2022 19:23:56 +0200
Subject: [PATCH 1/3] Allow up to v1.2 SPIR-V features

This is to support warp-level primitives (shuffles, ballots) via
extensions. The non-extension ones would require v1.3, but it's
not supported by any target yet.
---
 clang/lib/Driver/ToolChains/HIPSPV.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/clang/lib/Driver/ToolChains/HIPSPV.cpp b/clang/lib/Driver/ToolChains/HIPSPV.cpp
index bdbcf9109129..622497603292 100644
--- a/clang/lib/Driver/ToolChains/HIPSPV.cpp
+++ b/clang/lib/Driver/ToolChains/HIPSPV.cpp
@@ -96,8 +96,10 @@ void HIPSPV::Linker::constructLinkAndEmitSpirvCommand(
   }
 
   // Emit SPIR-V binary.
-
-  llvm::opt::ArgStringList TrArgs{"--spirv-max-version=1.1",
+  // We need 1.2 when using warp-level primitivies via sub group extensions.
+  // Strictly put we'd need 1.3 for the standard non-extension shuffle
+  // operations, but it's not supported by any target yet.
+  llvm::opt::ArgStringList TrArgs{"--spirv-max-version=1.2",
                                   "--spirv-ext=+all"};
   InputInfo TrInput = InputInfo(types::TY_LLVM_BC, TempFile, "");
   SPIRV::constructTranslateCommand(C, *this, JA, Output, TrInput, TrArgs);
-- 
2.43.0

